lib/Http/HtmlResponse.php

```diff
@@ -43,6 +48,6 @@ public function __construct(string $content) {
 	 * @return string the file contents
 	 */
 	public function render(): string {
-		return $this->content;
+		return '<style>' . $this->injectedStyles . '</style>' . $this->content;
```

Comment:
:thinking: Is this compliant to HTML standards? I think `$this->content` will most likely start with a `<html>`, right? So it might render because browsers are very flexible and fault tolerant, …

Reply:
Yeah it is not compliant at all, however HTML emails in general are in most cases doing weird stuff anyways trying to cover random email clients, so I'd rather stick with this simple noncompliant approach than introducing complex parsing logic that tries to figure out the head element or insert one if there isn't any.

Reply:
Fair enough :)

I did not mean to reject your solution, I just wanted to make sure we are aware that this could potentially cause issues depending on how flexible the rendering in the browser is.

Reply:
Just for reference, officially it could be put in head or body

https://www.w3.org/TR/html52/document-metadata.html#the-style-element
> Contexts in which this element can be used:
>     Where metadata content is expected. 
>     In a noscript element that is a child of a head element. 
>     In the body, where flow content is expected. https://www.w3.org/TR/html52/dom.html#flow-content